dask/array/reductions.py

```diff
@@ -614,7 +614,9 @@ def arg_reduction(x, chunk, combine, agg, axis=None, split_every=None, out=None)
                         "got '{0}'".format(axis))
 
     # Map chunk across all blocks
-    name = 'arg-reduce-chunk-{0}'.format(tokenize(chunk, axis))
+    name = 'arg-reduce-chunk-axis={0}-{1}'.format(axis, tokenize(x, chunk,
+                                                                 combine, split_every,
+                                                                 out))
```

Comment:
I recommend that we drop the `axis=` bit.  I don't think we want to include too much semantic information into these keys.  Also, should `agg` be included in this name?

Reply:
Ah I see, agg is used below in `_tree_reduce`.  Presumably we do want two computations with different agg's to share the first bit of the computation.

Reply:
remove `out` here as well then. I think `agg` should be included, but `chunks` not?

Reply:
I think that yes we should remove out.  I think that we need to include the `chunk` function, but not the `chunks` tuple (it is derived from other user provided quantities).  

I think that you were correct to omit `agg` for this part of the computation because `agg` is only used afterwards in the `_tree_reduce` line, which will include it there.

Reply:
Agreed, and I mistyped `chunks` for `chunk`. But do we need `"-chunk"` in the name?

Reply:
We do not need `-chunk` in the name.  The keys are useful in three ways: 

1.  Sometimes we debug raw task graphs, and it's useful to have informative names
2.  They show up in diagnostics, like the Dask dashboard or progress bars
3.  They are used by the distributed scheduler to guess at runtimes.  Keys that start the same are expected to take roughly the same time.

So we should design keys with those objectives in mind.  Users will see them in diagnostics, so they should be easy to understand, they should also probably be short-but-informative.